core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala

```diff
@@ -469,11 +469,11 @@ class GroupMetadataManager(brokerId: Int,
           if (isTxnOffsetCommit) {
             group.inLock {
               addProducerGroup(producerId, group.groupId)
-              group.prepareTxnOffsetCommit(producerId, offsetMetadata)
+              group.prepareTxnOffsetCommit(producerId, offsetMetadata.map { case(k, v) => k.topicPartition -> v })
             }
           } else {
             group.inLock {
-              group.prepareOffsetCommit(offsetMetadata)
+              group.prepareOffsetCommit(offsetMetadata.map { case(k, v) => k.topicPartition -> v })
```

Comment:
I didn't propagate the topic ids into `GroupMetadata` where the actual offset maps are. So, the conversion of keys from `TopicIdPartition` to `TopicPartition` happens here for both offset commit and transaction offset commit. Please let me know if we should go one level deeper and do this conversion in `GroupMetadata` itself (without changing the retained data).

Reply:
Sure - I pushed a new commit [b6d9d2b](https://github.com/apache/kafka/pull/13378/commits/b6d9d2ba247a7880002486b23bd86db63c03212e) which propagates `TopicIdPartition` to the following `GroupMetadata` methods:

```
onOffsetCommitAppend
prepareOffsetCommit
failPendingOffsetWrite
```

and:

```
onTxnOffsetCommitAppend
prepareTxnOffsetCommit
failPendingTxnOffsetCommit
```

Please note that not all public methods of `GroupMetadata` use `TopicIdPartition`, and some of them remain `TopicPartition`-based. Do you think it is fine? (since `GroupMetadata` is an internal class, it should be I suppose).

Reply:
Alternatively, a forward-looking approach would be to update all public methods of `GroupMetadata` to `TopicIdPartition` although that would mean drastically extending the scope to all callers beyond the offset commit path and using the zero topic id where required. I guess this wouldn't be advisable though?

Reply:
I would only change the ones related to the offset commit path (transactional and non-transactional) for now. Later one, I think that we may have to change the ones related to the fetch path as well because we will have the same issues.

Reply:
Makes sense!